### DISCUSSION TITLE
MapRequestPayload for PATCH (partial update)

### DISCUSSION THREAD

**Discussion by yblatti** [symfony/symfony#52127] [Q&A]
Hi everyone,

I'm writing a JSON API using `MapRequestPayload` on a DTO for `POST` and `PUT`.
And I'm having a blast, it works like a charm ! Thanks Symfony 6.3 contributors ❤️

My question is : can I use `MapRequestPayload` for partial update of an object using `PATCH`, sending only the fields I want to update ?

Thank you for your advice 🤗

**Comment by ruzampl** [comment]
Well, I'm looking for this too, but I think it's not a problem of MapRequestPayload and the mapped DTO, but what you do with that specific DTO. I think the correct way is to first populate the DTO with data from the entity (source data) and then map the property data from the request. So maybe custom request resolver is the way to go?

**Reply by yblatti** [reply]
Thanks for your reply.

I ended up doing something close to what you propose :

- Populate DTO from entity (`$myPopulatedDto`)
- Deserialize request Paylod (JSON) with Symfony Serializer, setting `AbstractNormalizer::OBJECT_TO_POPULATE => $myPopulatedDto`
- Validate `$myPopulatedDto` with Symfony Validator
- Write to entity

Clearly, the magic was not possible here using a DTO. My bad.

Thanks again for you reply.

**Comment by khasanshadiyarov** [comment]
What about nullable properties? E.g. a property needs to be set to null if it's specified as null, otherwise keep the old one.

**Reply by alexandre-le-borgne** [reply]
At the time I got away with wrapping the whole symfony denormalizer to save in a special property all the properties present in the request.
But that was before we could use this:
https://www.php.net/manual/en/reflectionproperty.isinitialized.php
Now it's probably easier to know if a property has been sent, just type it so it is initially uninitialized
